app/src/main/java/de/westnordost/streetcomplete/data/osm/osmquests/OsmQuest.kt

```diff
@@ -27,16 +28,26 @@ data class OsmQuest(
     override val type: QuestType<*> get() = osmElementQuestType
 
     override val markerLocations: Collection<LatLon> get() {
-        if (osmElementQuestType.hasMarkersAtEnds && geometry is ElementPolylinesGeometry) {
+        if (geometry is ElementPolylinesGeometry) {
             val polyline = geometry.polylines[0]
             val length = polyline.measuredLength()
-            if (length > 15 * 4) {
-                return listOf(
+            if (osmElementQuestType.hasMarkersAtEnds && length > 15 * 4) {
+                return kotlin.collections.listOf(
```

Comment:
```suggestion
                return listOf(
```

Reply:
I have tested it in action, but I haven't added any unit tests.

Reply:
suggested change no longer applies with the new position logic.